js/framework/AnalyticsConstants.js

```diff
@@ -9,37 +9,356 @@ if (!OO.Analytics.EVENTS)
+     * to load more data. It is also send when the stream is buffering before
+     * initial playback is started.
+     * @param {Array} paramArray Array of length 1, contains an instance of
+     * OO.Analytics.EVENT_DATA.VideoBufferingStartedData.
+     */
     VIDEO_BUFFERING_STARTED:        'video_buffering_started',
+
+    /**
+     * @public
+     * @event Analytics.EVENTS#VIDEO_BUFFERING_ENDED
+     * @description This message is sent when a video stream has buffered and
+     * is ready to resume playback.
+     * @param {Array} paramArray Array of length 1, contains an instance of
+     * OO.Analytics.EVENT_DATA.VideoBufferingEndedData.
+     */
     VIDEO_BUFFERING_ENDED:          'video_buffering_ended',
 
+    /**
+     * @public
+     * @event Analytics.EVENTS#AD_BREAK_STARTED
+     * @description This message is sent when the player stops the main content
+     * to start playing ads.
+     */
     AD_BREAK_STARTED:               'ad_break_started',
+
+    /**
+     * @public
+     * @event Analytics.EVENTS#AD_BREAK_ENDED
+     * @description This message is sent when the player has finished playing ads
+     * and is ready to playback the main video.
+     */
     AD_BREAK_ENDED:                 'ad_break_ended',
 
+    /**
+     * @public
+     * @event Analytics.EVENTS#DESTROY
+     * @description This message is sent when the player its plugins are destroying.
+     */
     DESTROY:                        'destroy'
   };
   OO.Analytics.EVENTS = EVENTS;
 }
 
+if (!OO.Analytics.EVENT_DATA)
+{
+  const EVENT_DATA = {};
+
+  /**
+   * @public
+   * @class Analytics.EVENT_DATA#VideoSourceData
+   * @classdesc Contains the data passed along with VIDEO_SOURCE_CHANGED. This
+   * includes the embed code (video id) and any metadata this video stream needs
+   * pass along to other plugins (Ex. It could contain ad tag data or analytics
+   * account information).
+   * @property  {string} embedCode This is the video stream's unique id
+   * @property  {object} metadata   An object containing metadata about the video stream and player id to be used.
+   */
+  EVENT_DATA.VideoSourceData = function(embedCode, metadata)
+  {
+    var checkSourceData = OO._.bind(checkDataType, this, "VideoSourceData");
+    this.embedCode = checkSourceData(embedCode, "embedCode", "string");
+    this.metadata  = checkSourceData(metadata, "metadata", "object");
+  }
+
+  /**
+   * @public
+   * @class Analytics.EVENT_DATA#VideoContentMetadata
+   * @classdesc Contains information about the content of the video stream,
+   * such as title and description.
+   * @property  {string} title Title of the video
+   * @property  {string} description Video description
+   * @property  {number} duration Duration of the video in milliseconds
+   * @property  {object} closedCaptions Object containing information about the closed captions available
+   * @property  {string} contentType A string indicating the type of content in the stream. (ex. "video")
```

Comment:
I don't know all the possibilities but I wanted to keep it open.  It the future there may be 'dvr' 'live' possibly something else.